client/modules/datafiles/src/DatafilesToolbar/DatafilesToolbar.tsx

```diff
@@ -7,12 +7,62 @@ import {
   USER_WORK_SYSTEM,
   useSelectedFiles,
   useSelectedFilesForSystem,
+  TFileListing,
 } from '@client/hooks';
 import DatafilesModal from '../DatafilesModal/DatafilesModal';
 import TrashButton from './TrashButton';
 import { Button, ButtonProps, ConfigProvider, ThemeConfig } from 'antd';
 import { useMatches, useParams } from 'react-router-dom';
 
+type ToolbarRules = {
+  canPreview: boolean;
+  canRename: boolean;
+  canCopy: boolean;
+  canMove: boolean;
+  canTrash: boolean;
+  canDownload: boolean;
+};
+
+/**
+ * Get the rules for enabling or disabling toolbar buttons based on the selected files, user permissions,
+ * system, and read-only mode.
+ */
+export function getToolbarRules(
+  selectedFiles: TFileListing[],
+  isReadOnly: boolean,
+  user: any,
```

Comment:
We can probably change this `user` param to a boolean called `isAuthenticated`. Then we can pass `!!user` as the argument to `getToolbarRules`

Reply:
done 👍